### PR TITLE
[Scss Grammar] Completed the major grammar refactoring

### PR DESCRIPTION
**Pull request by SergeyRusskih**
**Changes**

- Fixed major parsing errors and inconsistencies
- Removed Java test and replaced them with SASS files

### CODE REVIEW

**Comment by teverett** [comment]
Perhaps this?

```
Process is taking longer than 300 seconds. Killing process.
./scss test /Users/runner/work/grammars-v4/grammars-v4/scss/examples/bootstrap/_variables.scss failed

```

**Comment by SergeyRusskih** [comment]
Gotcha! Let me investigate the failures.

**Comment by SergeyRusskih** [comment]
Hi @KvanTTT ,

I've addressed the grammar and performance issues. Please, have a look.

**Comment by teverett** [comment]
@SergeyRusskih thanks!

File: scss/ScssParser.g4 [hunk 1]
```diff
@@ -31,393 +32,523 @@ parser grammar ScssParser;
...
-  ;
+parameter
+    : ( value | variableDeclaration | listSpaceSeparated | mapDeclaration ) arglist? prio?
+    ;
 
-keywordArgument
-  : identifierVariableName COLON expression
-  ;
+functionBody
+    : functionStatement* functionReturn?
```
Comment (KvanTTT) [inline scss/ScssParser.g4:336]: I have the following warning:

```
warning(154): ScssParser.g4:322:0: rule functionDeclaration contains an optional block with at least one alternative that can match an empty string
```

Please fix it by removing optionality from `functionBody`:

```suggestion
    : functionStatement+ functionReturn?
```

**Comment by KvanTTT** [comment]
@teverett could you mind not merging requests that need my review?

**Pull request merged**